docs/resources/dms_kafka_instance.md

```diff
@@ -0,0 +1,221 @@
+---
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# huaweicloud\_dms\_kafka\_instance
+
+## Example Usage
+
+### Basic Instance
+
+```hcl
+data "huaweicloud_dms_az" "test" {}
+
+data "huaweicloud_vpc" "test" {
+  name = "vpc-default"
+}
+
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+
+data "huaweicloud_dms_product" "test" {
+  engine        = "kafka"
+  instance_type = "cluster"
+  version       = "2.3.0"
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name        = "secgroup_1"
+  description = "secgroup for kafka"
+}
+
+resource "huaweicloud_dms_kafka_instance" "test" {
+  name                   = "instance_1"
+  description            = "kafka test"
+  access_user            = "user"
+  password               = "Kafkatest@123"
+  vpc_id                 = data.huaweicloud_vpc.test.id
+  subnet_id              = data.huaweicloud_vpc_subnet.test.id
+  security_group_id      = huaweicloud_networking_secgroup.test.id
+  available_zones        = [data.huaweicloud_dms_az.test.id]
+  product_id             = data.huaweicloud_dms_product.test.id
+  engine_version         = data.huaweicloud_dms_product.test.version
+  specification          = data.huaweicloud_dms_product.test.bandwidth
+  partition_num          = data.huaweicloud_dms_product.test.partition_num
+  storage_space          = data.huaweicloud_dms_product.test.storage
+  storage_spec_code      = data.huaweicloud_dms_product.test.storage_spec_code
+  kafka_manager_user     = "kafka-user"
+  kafka_manager_password = "Kafkatest@123"
+
+  tags = {
+    key   = "value"
+    owner = "terraform"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the DMS kafka instance resource.
+  If omitted, the provider-level region will be used. Changing this creates a new instance resource.
+
+* `name` - (Required, String) Specifies the name of the DMS kafka instance. An instance name starts with a letter,
+    consists of 4 to 64 characters, and supports only letters, digits, hyphens (-) and underscores (_).
+
+* `description` - (Optional, String) Specifies the description of the DMS kafka instance. It is a character
+    string containing not more than 1024 characters.
+
+* `engine_version` - (Required, String, ForceNew) Specifies the version of the kafka engine.
+    Valid values are "1.1.0" and "2.3.0".
+    Changing this creates a new instance resource.
+
+* `specification` - (Required, String, ForceNew) Specifies the baseline bandwidth of the DMS kafka instance,
```

Comment:
can we rename it to `bandwidth` to keep with the console?

Reply:
fixed